### ISSUE TITLE
Check if `requests` is installed?

### ISSUE THREAD

**Issue by Anselmoo** [mwouts/itables#123]
If `requests` is not pre-installed, the native installation (`pip install .`) can cause some trouble.

https://github.com/mwouts/itables/blob/d12e364690afb788947890080fa380af92eb3c4d/setup.py#L6

It might be a good idea to catch that via `try`-`except`?

## Solution 1:

```python
try:
    import requests
except ImportError:
    print("Please install requests")
    exit(1)
```

## Solution 2:

```python
try:
    import requests
except ImportError:
    import subprocess
    subprocess.check_call(['python', '-m', 'pip', 'install', 'requests'])
    import requests
```

**Comment by mwouts** [comment]
Hey @Anselmoo , maybe we could use a build requirement like documented here: https://setuptools.pypa.io/en/latest/userguide/dependency_management.html#build-system-requirement ? As `requests` is only used when the package is built... What do you think?

**Comment by Anselmoo** [comment]
That's an excellent idea!

However, [pyproject.toml](https://pip.pypa.io/en/stable/reference/build-system/pyproject-toml/) should also be considered used as a replacement for `setup.py` in the long term.

**Comment by Anselmoo** [comment]
@mwouts looks like it works, however I have to also fix the `flake8` issue #124
